permissions/src/iosMain/kotlin/PermissionTimerHelper.kt

```diff
@@ -26,35 +26,64 @@ import kotlinx.coroutines.sync.withLock
 
 class PermissionTimerHelper<P:Permission>(private val permissionManager: PermissionManager<P>, private val authorizationStatus: suspend () -> IOSPermissionsHelper.AuthorizationStatus, coroutineScope: CoroutineScope = permissionManager) : CoroutineScope by coroutineScope {
 
+    sealed class TimerJobState(internal val coroutineScope: CoroutineScope) {
+        class TimerNotRunning(coroutineScope: CoroutineScope): TimerJobState(coroutineScope) {
+            fun startTimer(interval: Long, block: suspend () -> Unit) = TimerRunning(interval, block, coroutineScope)
+        }
+
+        class TimerRunning(val interval: Long, val block: suspend () -> Unit, coroutineScope: CoroutineScope): TimerJobState(coroutineScope) {
+            private var timerLoop: Job
+
+            init {
+                timerLoop = createTimerLoop()
+            }
+
+            private fun createTimerLoop(): Job = coroutineScope.launch {
+                delay(interval)
+                block()
+                loopTimer()
```

Comment:
you can still just make a `while (true)` loop here (the loop gets escaped by the `CancellationException`). If you think that's ugly `while(isActive) {` would also do. I think it's much more expressive compared to relaunching itself every time.

Also timerLoop can become a `val` then.